syntax = "proto3";

import "google/protobuf/any.proto";
import "google/protobuf/empty.proto";
import "enums.proto";

package pslx;


service GenericRPCService {
    rpc RPCWithReturn (GenericRPCRequest)  returns (GenericRPCResponse) {}
    rpc RCPWithoutReturn (GenericRPCRequest) returns (google.protobuf.Empty) {}
}

// the next will be 5
message GenericRPCRequest {
    string timestamp = 1;
    string uuid = 2;
    google.protobuf.Any request_data = 3;
    Status status = 4;
}

// the next will be 5
message GenericRPCResponse {
    string timestamp = 1;
    string request_uuid = 2;
    google.protobuf.Any request_data = 3;
    Status status = 4;
}

